src/compiler/parser.ts

```diff
@@ -2704,6 +2704,14 @@ namespace ts {
                 return arrowExpression;
             }
 
+            // From ES6 spec:
+            // AsyncArrowFunction[In, Yield, Await]::
+            //      async[no LineTerminator here]AsyncArrowBindingIdentifier[?Yield][no LineTerminator here]=>AsyncConciseBody[?In]
+            const unParenthesizedAsyncArrowFunction = tryParseUnParenthesizedAsyncArrowFunctionExpressionWorker();

```

Comment:
help! it's the MicroSoft BiCapitalisation DisEase! 
should be `unparenthensizedAsyncArrowFunction` and `tryParseUnparenthesizedAsyncArrowFunctionExpressionWorker` :)